src/gpu/ocl/ref_sum.hpp

```diff
@@ -137,6 +137,9 @@ struct ref_sum_t : public gpu_primitive_t {
             r_ctx.set_scratchpad_grantor(ns.grantor());
             CHECK(reorders_[n]->execute(r_ctx));
         }
+#ifdef DNNL_SYCL_CUDA
+        ctx.stream()->wait();
+#endif
```

Comment:
What is the reason for the change? I can see that there is another such place in line 125. This call blocks execution until execution of all submitted kernels is completed. In the case when queue is out-of-order this can cause redundant synchronization. Is that expected?

Reply:
The issue seems to have been resolved through some of the changes implemented already so this PR is no longer needed. 
However, the wait on the lines above should probably stay because it's only called when it's not using the cuda backend

Reply:
Oh, I didn't notice it was an opencl primitive. The wait above should stay.